python/taichi/lang/matrix.py

```diff
@@ -403,6 +403,16 @@ class Matrix(TaichiOperations):
         0
     """
     _is_taichi_class = True
+    __array_priority__ = 1000
+
+    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
+        """Make sure operations between a numpy ndarray and Taichi matrix always return a Matrix.
+        This function is called only when the first operand is a numpy ndarray.
+        """
+        if method == "__call__":
+            name = ufunc.__name__
+            return getattr(self, f"__{name}__")(args[0].tolist(), **kwargs)
```

Comment:
would you mind adding a test with 2d numpy array with ti.Matrix?

Reply:
2d test added